src/Workspaces/Core/Portable/Workspace/Host/WorkspaceEventListener/IWorkspaceEventListenerProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// provide <see cref="IWorkspaceEventListener"/> for given workspace
+    /// </summary>
+    internal interface IWorkspaceEventListenerProvider : IWorkspaceService
+    {
+        IEnumerable<IWorkspaceEventListener> GetListeners();
```

Comment:
consdier being ImmutbaleArray.  IEnumerables always concern me (esp. in areas like this where i don't want this to be lazy, with work being duplciated each time the enumerable is iterated).

Reply:
removed.